pinot-core/src/main/java/org/apache/pinot/core/query/executor/QueryExecutor.java

```diff
@@ -54,6 +55,16 @@ void init(PinotConfiguration config, InstanceDataManager instanceDataManager, Se
 
   /**
    * Processes the query with the given executor service.
+   * <ul>
+   *   <li>
+   *     For streaming request, the returned DataTable contains only the metadata. The response is streamed back via the
+   *     observer.
+   *   </li>
+   *   <li>
+   *     For non-streaming request, the returned DataTable contains both data and metadata.
+   *   </li>
+   * </ul>
    */
-  DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService);
+  DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService executorService,
```

Comment:
Shall we have a different QueryExecutor implementation for streaming request rather than change this method in the interface?

Reply:
I agree. We should consider having a different implementation.